components/admin_console/permission_schemes_settings/edit_post_time_limit_modal/edit_post_time_limit_modal.tsx

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+import {Modal} from 'react-bootstrap';
+
+import {AdminConfig} from 'mattermost-redux/types/config';
+import {ActionFunc} from 'mattermost-redux/types/actions';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import {Constants} from 'utils/constants';
+import {localizeMessage} from 'utils/utils.jsx';
+import {t} from 'utils/i18n';
+
+const INT32_MAX = 2147483647;
+
+type Props ={
+    config: Partial<AdminConfig>;
+    show: boolean;
+    onClose: () => void;
+    actions: {
+        updateConfig: (config: AdminConfig) => ActionFunc & Partial<{error?: ClientErrorPlaceholder}>;
+        getConfig: () => ActionFunc;
+    };
+}
+
+type ClientErrorPlaceholder = {
+    message: string;
+    server_error_id: string;
+}
+
+export default function EditPostTimeLimitModal(props: Props) {
+    const [saving, setSaving] = React.useState(false);
+    const [errorMessage, setErrorMessage] = React.useState('');
+    const [postEditTimeLimit, setPostEditTimeLimit] = React.useState<number | string>(parseInt(props.config.ServiceSettings!.PostEditTimeLimit.toString(), 10));
```

Comment:
I was converting it to string because in the old version the value was set to `number` or `string` in the following use cases
1. set it to number -1 for always edit or bigger than 0 to define a certain number of seconds 
2. set it to empty string ('') on `handleOptionChange`

Anyhow with the logic you proposed, we don't need it anymore so the type can only be `number`